### PR TITLE
add waitforTransform to prevent lookupexception

### LINKED ISSUE #522: Configuring ED failed
I assume that it is caused by TF listeners in ED plugins not being handled correctly. I will put some effort in it today, 2017-13-06.
```
terminate called after throwing an instance of 'tf2::LookupException'
  what():  "map" passed to lookupTransform argument target_frame does not exist. 
/home/amigo/ros/kinetic/system/devel/lib/ed/ed(_Z13signalHandleri+0x558)[0x4dbd17]
/lib/x86_64-linux-gnu/libc.so.6(+0x354b0)[0x7f029a7754b0]
/lib/x86_64-linux-gnu/libc.so.6(gsignal+0x38)[0x7f029a775428]
/lib/x86_64-linux-gnu/libc.so.6(abort+0x16a)[0x7f029a77702a]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(_ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x16d)[0x7f029adaf84d]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8d6b6)[0x7f029adad6b6]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(+0x8c6a9)[0x7f029adac6a9]
/usr/lib/x86_64-linux-gnu/libstdc++.so.6(__gxx_personality_v0+0x2e5)[0x7f029adad005]
/lib/x86_64-linux-gnu/libgcc_s.so.1(+0xff83)[0x7f029ab19f83]
/lib/x86_64-linux-gnu/libgcc_s.so.1(_Unwind_Resume+0x57)[0x7f029ab1a487]
[ERROR] [1512505793.347695]: Could not configure ED:

Service call failed: transport error completing service call: unable to receive data from sender, check sender's logs for details
[amigo/ed_configure-4] process has died [pid 11714, exit code 1, cmd /home/amigo/ros/kinetic/system/src/ed/tools/configure /home/amigo/ros/kinetic/system/src/amigo_bringup/parameters/world_modeling/world_model.yaml __name:=ed_configure __log:=/home/amigo/.ros/log/c607402c-d9c1-11e7-9bb2-00031d0a0d6c/amigo-ed_configure-4.log].
log file: /home/amigo/.ros/log/c607402c-d9c1-11e7-9bb2-00031d0a0d6c/amigo-ed_configure-4*.log
[amigo/ed-3] process has died [pid 11709, exit code 1, cmd /home/amigo/ros/kinetic/system/devel/lib/ed/ed __name:=ed __log:=/home/amigo/.ros/log/c607402c-d9c1-11e7-9bb2-00031d0a0d6c/amigo-ed-3.log].
log file: /home/amigo/.ros/log/c607402c-d9c1-11e7-9bb2-00031d0a0d6c/amigo-ed-3*.log
```

### PR DESCRIPTION
**Pull request by MatthijsBurgh**
This should fix tue-robotics/tue_robocup#522, ED crashing during configure.

An LoopUpException is raised. in the lookuptransform. While the transform is available. It does give the error shown in tue-robotics/tue_robocup#522. Adding the waitfortransform already solved the issue in my opinion, it didn't show up, when restarting free-mode ~15 times. To make it more robust I added a try/catch.

How to test: Start amigo-free-mode on AMIGO multiple times.

### CODE REVIEW

**Comment by Rayman** [comment]
So if there is no transform, the localization plugin will WAIT in the loop. That looks not like good behavior. You can for example catch `tf::LookupException` and then log & return.

**Comment by MatthijsBurgh** [comment]
It is only during the configure, so not in the update loop. Normally there is a transform available, so then it shouldn't take time. So maybe it is a good idea to reduce the timeout to 1 (or smaller) sec. I think that is allowable.

**Comment by MatthijsBurgh** [comment]
@Rayman, could you please review this one again?

**Comment by MatthijsBurgh** [comment]
Ping

File: src/localization_plugin.cpp [hunk 1]
```diff
@@ -115,18 +115,31 @@ void LocalizationPlugin::configure(tue::Configuration config)
...
-        ROS_DEBUG_STREAM("Initial pose from parameter server: [" << p.x << ", " << p.y << "], yaw:" << yaw);
+                ROS_DEBUG_STREAM("Initial pose from parameter server: [" << p.x << ", " << p.y << "], yaw:" << yaw);
+            }
+            catch (tf::TransformException ex)
+            {
+                ROS_ERROR("[ED Localization] %s",ex.what());
+            }
+        }
+        else
+            ROS_ERROR("[ED Localization] no transform between odom and base_link.");
```
Comment (jlunenburg) [inline src/localization_plugin.cpp:144]: Cleanup up some error messages and comment style. Approved by @Rayman

**Review by jlunenburg — APPROVED** [review]

**Pull request merged**